pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1469,30 +1463,42 @@ func (r *ReconcileClusterDeployment) ensureManagedDNSZone(cd *hivev1.ClusterDepl
 
 	if !metav1.IsControlledBy(dnsZone, cd) {
 		cdLog.Error("DNS zone already exists but is not owned by cluster deployment")
-		if err := r.setDNSNotReadyCondition(cd, false, "Existing DNS zone not owned by cluster deployment", cdLog); err != nil {
+		if err := r.setDNSNotReadyCondition(cd, corev1.ConditionTrue, dnsNotReadyReason, "Existing DNS zone not owned by cluster deployment", cdLog); err != nil {
 			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not update DNSNotReadyCondition")
 			return nil, err
 		}
 		return nil, errors.New("Existing unowned DNS zone")
 	}
 
 	availableCondition := controllerutils.FindDNSZoneCondition(dnsZone.Status.Conditions, hivev1.ZoneAvailableDNSZoneCondition)
-	if availableCondition == nil || availableCondition.Status != corev1.ConditionTrue {
-		// The clusterdeployment will be queued when the owned DNSZone's status
-		// is updated to available.
-		cdLog.Debug("DNSZone is not yet available. Waiting for zone to become available.")
-		if err := r.setDNSNotReadyCondition(cd, false, "DNS Zone not yet available", cdLog); err != nil {
-			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not update DNSNotReadyCondition")
-			return nil, err
-		}
-		return nil, nil
+	insufficientCredentialsCondition := controllerutils.FindDNSZoneCondition(dnsZone.Status.Conditions, hivev1.InsufficientCredentialsCondition)
+	var (
+		status          corev1.ConditionStatus
+		reason, message string
+	)
+	switch {
+	case availableCondition != nil && availableCondition.Status == corev1.ConditionTrue:
+		status = corev1.ConditionFalse
+		reason = dnsReadyReason
+		message = "DNS Zone available"
+	case insufficientCredentialsCondition != nil && insufficientCredentialsCondition.Status == corev1.ConditionTrue:
+		status = corev1.ConditionTrue
+		reason = "InsufficientCredentials"
+		message = insufficientCredentialsCondition.Message
+	default:
+		status = corev1.ConditionTrue
+		reason = dnsNotReadyReason
+		message = "DNS Zone not yet available"
 	}
-
-	if err := r.setDNSNotReadyCondition(cd, true, "DNS Zone available", cdLog); err != nil {
+	if err := r.setDNSNotReadyCondition(cd, status, reason, message, cdLog); err != nil {
 		cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not update DNSNotReadyCondition")
 		return nil, err
 	}
-	return dnsZone, nil
+
+	if reason == dnsReadyReason {
+		return dnsZone, nil
+	}
+	return nil, nil
```

Comment:
This is a nit, but I would rather the return at the end of the function be the happy scenario.
```suggestion
	if reason != dnsReadyReason {
		return nil, nil
	}
	return dnsZone, nil
```